File: wiringPi/python/blink.py
#!/usr/bin/env python

'''
Program: blink.py
Description: Blinking led
			 using wiringPi library
Author: Aldo Nunez
'''

# Import GPIO module
import wiringpi as wipi
import time 

def main ():
	# setup wiringpi
	wipi.wiringPiSetup ()

	# wPi 9 corresponds to Pin 5
	WPI_PORT = 9;

	# time delay in milliseconds
	T = 0.05 	# 0.05 ~ 50 ms


	# '1' means output 
 	wipi.pinMode ( WPI_PORT, 1 )

	# initial state
	state = 1
	print ( "blinking Pin 5" )
	while True:
		# blink
		wipi.digitalWrite ( WPI_PORT, state )

		# waiting
		time.sleep ( T )

		# switch state
		state = not state

if __name__  == "__main__":
	main ()

File: wiringPi/c++/src/blink.cpp
//
// Program: blink.cpp
// Purpose: Blinks led conncected to Pin 5
//          using wiringPi library
// Author:  Aldo Nunez
//

#include <iostream>
#include <wiringPi.h>

using namespace std;

int
main ( void )
{
    // setup 
    if ( wiringPiSetup () )
    {
        cerr <<   "Failing to setup wiringPi" << endl;
        return 1;
    }

    // wPi 9 corresponds to Pin 5 
    const int WPI_PORT = 9;

    // delay in milliseconds 
    const int T = 50; 	

    // sets the port as output
    pinMode ( WPI_PORT, OUTPUT );

    // initial state 
    unsigned short int state = HIGH;

    cout << "blinking Pin 5" << endl;
    while ( 1 )
    {
        // blink 
        digitalWrite ( WPI_PORT, state );

        // waiting 
        delay ( T );

        // switch state
        state = !state;
    } // end of while

    return 0;
}

